src/main.cpp

```diff
@@ -658,14 +658,8 @@ bool IsStandardTx(const CTransaction& tx, string& reason)
 
 bool IsFinalTx(const CTransaction &tx, int nBlockHeight, int64_t nBlockTime)
 {
-    AssertLockHeld(cs_main);
-    // Time based nLockTime implemented in 0.1.6
     if (tx.nLockTime == 0)
         return true;
-    if (nBlockHeight == 0)
```

Comment:
> I am quite sure of that, particularly since the callers that would do that are all consensus-critical code, and this could lead to a consensus split. Edit: maybe that comes off the wrong way... I mean, because it could lead to a consensus split, I spent a good hour or so examining every call of IsFinalTx() to be sure it'd never be called that way.

Then why maintain a global-state-dependent function at all?
If we're going to review all the calls and change all those lines, let's just have the stateless version of it.
And at that point, you're only changing one function, shouldn't the criterion "if you change functionality, change the name of the function" precisely apply here?
I'll elaborate on the main thread...

> Fortunately the only block for which nBlockHeight=0 or nBlockTime=0 is legal is the genesis block so we don't actually have that bug in practice. 

Well, the genesis block shouldn't be tested since it's correct by definition (is really the first consensus rule), so that's fine.

> Additionally if nBlockHeight/nBlockTime are set to zero the function still behaves as you'd expect it to in that circumstance.

No, before the null values were replaced with values from the global state, now they don't.
They clearly function differently (even if you're taking care of it not mattering by adapting things on the caller, the function itself is NOT functionally equivalent to the previous stateful one).